src/components/oscal-utils/OSCALLinkUtils.js

```diff
@@ -1,18 +1,10 @@
 import getUriFromBackMatterByHref from "./OSCALBackMatterUtils";
 
-export function fixJsonUrls(absoluteUrl) {
-  // TODO: This workaround references JSON representation instead of the back-matter links to
-  // XML catalogs and it should be removed once the issue has been resolved.
-  // https://github.com/EasyDynamics/oscal-react-library/issues/158
-  if (!absoluteUrl.endsWith(".xml")) {
-    return absoluteUrl;
-  }
-  // Replacing all instances of xml with json in the path *should* get us the correct json URL
-  return absoluteUrl.replace(/xml/g, "json");
-}
-
 export function getAbsoluteUrl(href, parentUrl) {
-  return href.startsWith("http") ? href : `${parentUrl}/../${href}`;
+  if (href.startsWith("https://") || href.startsWith("http://") || !parentUrl) {
+    return href;
+  }
+  return new URL(href, parentUrl).toString();
```

Comment:
Seems URL() fails when `parentUrl` is a relative link (eg. when the object in backMatter points to a relative link as its `rlink` parameter)
For example, when I run a Profile that contains a relative link, the console shows the error: 
`TypeError: Failed to construct 'URL': Invalid base URL
(OSCALLinkUtils.js:7:10)`
[Reference docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) 

We probably need another conditional here to check whether `parentUrl` is a relative link, and if so resolve the link correctly.

This overlaps with [github.com/EasyDynamics/oscal-editor-deployment/issues/141](https://github.com/EasyDynamics/oscal-editor-deployment/issues/141)

Reply:
Actually, thinking about this a little more, we could require that `parentUrl` is an absolute link as it is created. This variable originates from the environment variable `REST_APP_BASE_URL` that is set by the user.
An important case when this environment variable must be a relative link is in the OSCAL Editor Docker Deployment; the relative link is used so that the Docker container operates agnostic of the port the user chooses.

Seems we either need a check here in `OSCALLinkUtils.js` to fix the URL in the case of a relative link,
or to add a check elsewhere in the code as `REST_APP_BASE_URL` is initially resolved to grab the user's specified port and build an absolute URL.
We could also rethink the use of the environment variable at build time for this purpose if it's too clunky, but if I recall correctly we didn't find a nice way of specifying the base REST URL in the container any other way.

Would like to see everyone's thoughts on a good approach discussed here: https://github.com/EasyDynamics/oscal-editor-deployment/issues/141